ComponentKit/Core/CKComponentMemoizer.mm

```diff
@@ -149,6 +185,11 @@ id CKMemoize(CKMemoizationKey memoizationKey, id (^block)(void))
 {
   _CKComponentMemoizerImpl *impl = [_CKComponentMemoizerImpl currentMemoizer];
   CKComponent *component = [impl dequeueComponentForKey:memoizationKey];
+  if (component) {
+    if (!CKCanUseMemoizedComponent(component)) {
```

Comment:
can we just move this check inside `dequeue`? Or do you prefer to keep it here for cleanliness of the queue? your call